openfl-tutorials/experimental/106_Scikit_Learn_Linear_Regression_Workflow.ipynb

```diff
@@ -0,0 +1,551 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "9a61e30e-83d1-422d-aa0a-ebd4376197ec",
+   "metadata": {},
+   "source": [
+    "# Scikit-learn Linear Regression Tutorial using Workflow Interface with Ridge Regularization\n",
+    "\n",
+    "\n",
+    "This tutorial demonstrates how to train a linear regression model using scikit-learn with Ridge regularization on a dataset, leveraging the new FedAI Workflow Interface. The Workflow Interface provides a novel way to compose federated learning experiments with OpenFL, enabling researchers to handle non-IID data and perform federated averaging. Through this tutorial, you will learn how to set up the federated learning environment, define the flow, and execute the training process across multiple collaborators."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c4b334ef-6a72-4b82-b978-1401973d0512",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## We will use MSE as loss function and Ridge weights regularization\n",
+    "![image.png](https://www.analyticsvidhya.com/wp-content/uploads/2016/01/eq5-1.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b9e4fca5-559a-4fe7-9d3a-8f365637a0dd",
+   "metadata": {},
+   "source": [
+    "## What is it?\n",
+    "\n",
+    "The Workflow Interface is a new way of composing federated learning experiments with OpenFL. It was developed through conversations with researchers and existing users who had novel use cases that didn't quite fit the standard horizontal federated learning paradigm."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1cef8056-1d16-4157-8686-7e2165d570bb",
+   "metadata": {},
+   "source": [
+    "## Getting Started"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2843ae32-23b7-4323-a468-0426c9daabf0",
+   "metadata": {},
+   "source": [
+    "First we start by installing the necessary dependencies for the workflow interface"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "689ee822",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install -r workflow_interface_requirements.txt"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3b436c8b",
+   "metadata": {},
+   "source": [
+    "Now, we import the relevant modules and do some basic initializations"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6c9eee14-22a1-4d48-a7da-e68d01037cd4",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "from typing import List, Union\n",
+    "\n",
+    "from sklearn.linear_model import Lasso\n",
+    "from sklearn.preprocessing import StandardScaler\n",
+    "from sklearn.datasets import make_regression\n",
```

Comment:
not being used

Reply:
resolved in e1e711efeb4f84ed325d0ae3401328c80502479a